drivers/rtt_rtc/rtt_rtc.c

```diff
@@ -50,7 +51,7 @@
 #define BACKUP_RAM      __attribute__((section(".noinit")))
 #endif
 
-static uint32_t rtc_now BACKUP_RAM;         /**< The RTC timestamp when the last RTT alarm triggered */
+static volatile uint32_t rtc_now BACKUP_RAM; /**< The RTC timestamp when the last RTT alarm triggered */
```

Comment:
I really dislike using `volatile` on a variable that does not refer to memory-mapped I/O.

1. It feeds into the misinformation campaign that `volatile` magically solves all data race issues that is commonly spread by embedded developers
2. It results in more overhead in both terms of ROM and CPU cycles, as `volatile` really only is about disabling optimizations. E.g. withing `irq_disable()` ... `irq_restore()`, memory mapped I/O register still can change (hence, every access is `volatile`), but ordinary SRAM will not magically change.
    - To some part this is IMO an oversight of the C spec, as it would have made much more sense to specify for specific accesses to be `volatile` rather than making this a type qualifier.
3. It is extremely hard to reason when one really only needs `volatile`, or also atomic access or sequential consistency. And this has to be done for *every* access to the variable. It is extremely easy to overlook an instance where more than just `volatile` is needed or to introduce bugs in refactoring. And these data-race bugs are a pain in the ass to debug.

Reply:
I recon this is an opportunity to make use of `atomic_load_u32()` :wink: